app/main.py

```diff
@@ -1,8 +1,35 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self, comfort_class: int,
+                 clean_mark: int, brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self, distance_from_city_center: int, clean_power: int,
+                 average_rating: float, count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list) -> float:
+        income = 0.0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                income += self.calculate_washing_price(car)
+            self.wash_single_car(car)
+        self.rate_service(5)
+        return income
+
+    def calculate_washing_price(self, car: Car) -> float:
+        return round((car.comfort_class * (self.clean_power - car.clean_mark) * self.average_rating / self.distance_from_city_center), 1)
+
+    def wash_single_car(self, car: Car) -> None:
+        if car.clean_mark < self.clean_power:
+            car.clean_mark = self.clean_power
+
+    def rate_service(self, i: int) -> None:
+        self.average_rating += 0.1
```

Comment:
The logic for updating `average_rating` is incorrect. Simply adding 0.1 does not correctly update the average rating based on a new rating. You should calculate the new average using the formula: `new_average = ((old_average * count_of_ratings) + new_rating) / (count_of_ratings + 1)`.

Reply:
Thank you